File: src/main/java/ru/hollowhorizon/mastertech/proxy/ClientProxy.java
package ru.hollowhorizon.mastertech.proxy;

import net.minecraft.client.renderer.block.model.ModelResourceLocation;
import net.minecraft.item.Item;
import net.minecraftforge.client.model.ModelLoader;
import net.minecraftforge.fml.common.event.FMLInitializationEvent;
import net.minecraftforge.fml.common.event.FMLPostInitializationEvent;
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
import ru.hollowhorizon.mastertech.api.IProxy;

import java.util.Objects;

public class ClientProxy implements IProxy {
    @Override
    public void preInit(FMLPreInitializationEvent e) {}

    @Override
    public void init(FMLInitializationEvent e) {}

    @Override
    public void postInit(FMLPostInitializationEvent e) {}

    @Override
    public void getRender(Item item, int meta, String id) {
        ModelLoader.setCustomModelResourceLocation(item, meta, new ModelResourceLocation(Objects.requireNonNull(item.getRegistryName()), id));
    }
}

File: src/main/java/ru/hollowhorizon/mastertech/proxy/ClientProxy.kt
package ru.hollowhorizon.mastertech.proxy

import net.minecraft.client.renderer.block.model.ModelResourceLocation
import net.minecraft.item.Item
import net.minecraftforge.client.model.ModelLoader
import net.minecraftforge.fml.common.event.FMLInitializationEvent
import net.minecraftforge.fml.common.event.FMLPostInitializationEvent
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent
import ru.hollowhorizon.mastertech.api.IProxy

class ClientProxy : IProxy {
    override fun preInit(e: FMLPreInitializationEvent) {}

    override fun init(e: FMLInitializationEvent) {}

    override fun postInit(e: FMLPostInitializationEvent) {}

    override fun getRender(item: Item, meta: Int, id: String) {
        item.registryName?.let { ModelResourceLocation(it, id) }?.let {
            ModelLoader.setCustomModelResourceLocation(
                item,
                meta,
                it
            )
        }
    }
}
